docs/en/sql-reference/functions/uuid-functions.md

```diff
@@ -18,39 +18,213 @@ generateUUIDv4([x])
 
 **Arguments**
 
-- `x` — [Expression](../../sql-reference/syntax.md#syntax-expressions) resulting in any of the [supported data types](../../sql-reference/data-types/index.md#data_types). The resulting value is discarded, but the expression itself if used for bypassing [common subexpression elimination](../../sql-reference/functions/index.md#common-subexpression-elimination) if the function is called multiple times in one query. Optional parameter.
+- `x` — [Expression](../../sql-reference/syntax.md#syntax-expressions) resulting in any of the [supported data types](../../sql-reference/data-types/index.md#data_types). The expression is used to bypass [common subexpression elimination](../../sql-reference/functions/index.md#common-subexpression-elimination) if the function is called multiple times in a query but otherwise ignored. Optional.
 
 **Returned value**
 
-The UUID type value.
+A value of type UUIDv4.
 
-**Usage example**
+**Example**
 
-This example demonstrates creating a table with the UUID type column and inserting a value into the table.
+First, create a table with a column of type UUID, then insert a generated UUIDv4 into the table.
 
 ``` sql
-CREATE TABLE t_uuid (x UUID) ENGINE=TinyLog
+CREATE TABLE tab (uuid UUID) ENGINE = Memory;
 
-INSERT INTO t_uuid SELECT generateUUIDv4()
+INSERT INTO tab SELECT generateUUIDv4();
 
-SELECT * FROM t_uuid
+SELECT * FROM tab;
 ```
 
 ```response
-┌────────────────────────────────────x─┐
+┌─────────────────────────────────uuid─┐
 │ f4bf890f-f9dc-4332-ad5c-0c18e73f28e9 │
 └──────────────────────────────────────┘
 ```
 
-**Usage example if it is needed to generate multiple values in one row**
+**Example where multiple UUIDs are generated per row**
 
 ```sql
-SELECT generateUUIDv4(1), generateUUIDv4(2)
+SELECT generateUUIDv4(1), generateUUIDv4(2);
+
 ┌─generateUUIDv4(1)────────────────────┬─generateUUIDv4(2)────────────────────┐
 │ 2d49dc6e-ddce-4cd0-afb8-790956df54c1 │ 8abf8c13-7dea-4fdf-af3e-0e18767770e6 │
 └──────────────────────────────────────┴──────────────────────────────────────┘
 ```
 
+## generateUUIDv7
+
+Generates a [UUID](../data-types/uuid.md) of [version 7](https://datatracker.ietf.org/doc/html/draft-peabody-dispatch-new-uuid-format-04).
+
+The generated UUID contains the current Unix timestamp in milliseconds (48 bits), followed by version "7" (4 bits), a counter (42 bit) to distinguish UUIDs within a millisecond (including a variant field "2", 2 bit), and a random field (32 bits).
```

Comment:
For a casual reader, it is not immediately clear how `generateUUIDv7` is different from the other two variants. A sentence analogous to l. 122/123 would be helpful.